### PR TITLE
Editorial review of the V1.0 book text.

### PR DESCRIPTION
**Pull request by jvanderk**
Editorial change proposals for V1.0 book text, English version only.

### CODE REVIEW

File: book/en/src/by-example/app_priorities.md [hunk 1]
```diff
@@ -18,21 +18,21 @@ The highest static priority task takes precedence when more than one
...
-3 │                      Preempts                          │
-2 │                    A─────────►                         │
-1 │          B─────────► - - - - B────────►                │
-0 │Idle┌─────►                   Resumes  ┌──────────►     │
+  │                                                                                           │
+  │                                                                                           │
+3 │                                   Preempts                                                │
+2 │                                 A─────────►                                         │
+1 │                 B─────────► - - - - - - - B────────►                           │
+0 │   Idle┌─────►                                    Resumes  ┌──────────►         │
```
Comment (jvanderk) [inline book/en/src/by-example/app_priorities.md:33]: I'm afraid my editor environment has hashed this up.

File: book/en/src/internals/conceptual_view.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# Conceptual view
+
+[Conceptual execution model](./conceptual_execution_model.png)
```
Comment (jvanderk) [inline book/en/src/internals/conceptual_view.md:3]: File name of png is incorrect.

File: book/en/src/internals/conceptual_view.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+One queue is a software queue, which is a a set of FIFOs written in
+code and managed by the RTIC framework. Individual FIFOs are
+associated with different execution priority levels.
+
+The other is a hardware-controlled queue maintained by the NVIC. If
+there are two hardware interrupts pending, the highest priority code
+will execute first: a "priority queue" at a hardware level.
+
+The the NVIC operates essentially independently of the CPU, outside of
```
Comment (jvanderk) [inline book/en/src/internals/conceptual_view.md:21]: Duplicate 'the'

Comment (AfoHT) [inline book/en/src/internals/conceptual_view.md:21]:
```suggestion
The NVIC operates essentially independently of the CPU, outside of
```

File: book/en/src/by-example/app_priorities.md [hunk 4]
```diff
@@ -4,7 +4,7 @@
 
 The `priority` argument declares the static priority of each `task`.
 
-For Cortex-M, tasks can have priorities in the range `1..=(1 << NVIC_PRIO_BITS)`
+For Cortex-M, tasks can have priorities in the range `1..(1 << NVIC_PRIO_BITS)`
```
Comment (AfoHT) [inline book/en/src/by-example/app_priorities.md:7]: In Rust, `..=` indicates it is an inclusive range

**Review by AfoHT — COMMENTED** [review]
Thank you for taking the time to go through this!

I've commented on some of the smaller things, need to further study the part you've added to internal docs.

A hint, for example in the case where you had duplicate "the", you can push more commits to this branch which will update the Pull Request accordingly :+1:

File: book/en/src/by-example/app_priorities.md (hunk 4 above)
Comment (AfoHT) [reply book/en/src/by-example/app_priorities.md:7]: https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html

Comment (perlindgren) [reply book/en/src/by-example/app_priorities.md:7]: If I recall correctly that is intentional, we can provide one more priority level by disabling the interrupts globally.

Comment (jvanderk) [reply book/en/src/by-example/app_priorities.md:7]: I was not aware of the rangeinclusive notation. Thx.

File: book/en/src/by-example/app_priorities.md [hunk 5]
```diff
@@ -53,7 +53,8 @@ when `baz`returns. When `bar` returns `foo` can resume.
 
 One more note about priorities: choosing a priority higher than what the device
 supports will result in a compilation error.
-The error is cryptic due to limitations in the language,
+
+The error is cryptic due to limitations in the language (XXX  which language ? XXX),
```
[the remark this answers is not in the document]
Comment (jvanderk) [reply book/en/src/by-example/app_priorities.md:57]: Clear.

File: book/en/src/SUMMARY.md [hunk 6]
```diff
@@ -37,3 +38,6 @@
   <!--- [Ceiling analysis](./internals/ceilings.md)-->
   <!--- [Software tasks](./internals/tasks.md)-->
   <!--- [Timer queue](./internals/timer-queue.md)-->
+
+(XXX would it be a good idea to add a section on code organization,
```
[the remark this answers is not in the document]
Comment (jvanderk) [reply book/en/src/SUMMARY.md:42]: The syd app I constructed is a rudimentary working example.

File: book/en/src/by-example/hardware_tasks.md [hunk 7]
```diff
@@ -1,24 +1,25 @@
...
+All tasks bound to an explicit interrupt are called *hardware tasks* since they
 start execution in reaction to a hardware event.
 
 Specifying a non-existing interrupt name will cause a compilation error. The interrupt names
 are commonly defined by [PAC or HAL][pacorhal] crates.
 
-Any available interrupt vector should work, but different hardware might have
-added special properties to select interrupt priority levels, such as the
+Any available interrupt vector should work. Specific devices may
+allow selection of interrupt priority level, such as the
```
[the remark this answers is not in the document]
Comment (jvanderk) [reply book/en/src/by-example/hardware_tasks.md:19]: It was not clear to me what the message of the original text was. "added properties to select ... priority" -> "to allow selection of" was intended to mean the same and break the original sentence in distinct statements. Reading your comment I would word this possibly as "Specific hardware may bind specific interrupt priorities to specific interrupt vectors outside user code control." Or something like that...

**Comment by jvanderk** [comment]
I'm struggling with magit---pushing commits to [jvanderk/cortex-m-rtic](https://github.com/jvanderk/cortex-m-rtic), but I'm not sure they are visible in the pull request.

File: book/en/src/by-example/app_init.md [hunk 8]
```diff
@@ -1,21 +1,23 @@
...
-signature `fn(init::Context) -> (Shared, Local, init::Monotonics)`, where `Shared` and `Local` are the resource
+signature `fn(init::Context) -> (Shared, Local, init::Monotonics)`, where `Shared` and `Local` are resource
 structures defined by the user.
 
-The `init` task executes after system reset (after the optionally defined `pre-init` and internal RTIC
-initialization). The `init` task runs *with interrupts disabled* and has exclusive access to Cortex-M (the
-`bare_metal::CriticalSection` token is available as `cs`) while device specific peripherals are available through
-the `core` and `device` fields of `init::Context`.
+The `init` task executes after system reset (after an optionally defined `pre-init` task/code section(?) 
+and always occurring internal RTIC
```
[the remark this answers is not in the document]
Comment (jvanderk) [reply book/en/src/by-example/app_init.md:8]: Don't understand the suggestion.

[the remark this answers is not in the document]
Comment (AfoHT) [reply book/en/src/by-example/app_init.md:8]: I interpreted the (?) after "task/code section(?)" as seeking clarification / sources, my thought was that the link could be useful here, the link in the comment.

File: book/en/src/by-example/hardware_tasks.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (AfoHT) [reply book/en/src/by-example/hardware_tasks.md:19]: > It was not clear to me what the message of the original text was. "added properties to select ... priority" -> "to allow selection of" was intended to mean the same and break the original sentence in distinct statements. Reading your comment I would word this possibly as "Specific hardware may bind specific interrupt priorities to specific interrupt vectors outside user code control." Or something like that...

^ Sounds good :+1:

File: book/en/src/internals/conceptual_view.md (hunk 2 above)
Comment (jvanderk) [reply book/en/src/internals/conceptual_view.md]: Corrected.

File: book/en/src/internals/conceptual_view.md (hunk 3 above)
Comment (jvanderk) [reply book/en/src/internals/conceptual_view.md]: Corrected.

File: book/en/src/by-example/app_init.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (jvanderk) [reply book/en/src/by-example/app_init.md]: Clear, added suggestion to text. Please check, never used that syntax.

File: book/en/src/by-example/app_priorities.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jvanderk) [reply book/en/src/by-example/app_priorities.md]: Resolved.

**Comment by jvanderk** [comment]
Please let me know if there are still open items.

I am working on getting vigilant mode working. Apologies for 'unverified' flags.

File: book/en/src/by-example/hardware_tasks.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (jvanderk) [reply book/en/src/by-example/hardware_tasks.md]: Amended text.

File: book/en/src/by-example/app_priorities.md [hunk 9]
```diff
@@ -18,11 +18,12 @@ The highest static priority task takes precedence when more than one
...
 
 The following scenario demonstrates task prioritization:
-Spawning a higher priority task A during execution of a lower priority task B pends
-task A. Task A has higher priority thus preempting task B which gets suspended
+Spawning a higher priority task A during execution of a lower priority task B suspends
+task B. Task A has higher priority thus preempting task B which gets suspended
 until task A completes execution. Thus, when task A completes task B resumes execution.
 
 ```text
+Task Priority
```
Comment (AfoHT) [inline book/en/src/by-example/app_priorities.md:26]: Duplicated lines?

File: book/en/src/internals/conceptual_view.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+Interrupt-driven
+Concurrency](https://rtic.rs/0.5/book/en/internals/tasks.html).
+
+## spawning
+
+Any task may influence the execution context via the "spawn" API,
+adding tasks to the queues for later execution.
+
+The defined priority of the task spawned determines the queue it lands
+in.
```
Comment (AfoHT) [inline book/en/src/internals/conceptual_view.md:80]: I think this would be best split out into a separate PR, as this is not part of the editorial review but entirely new content.

Included in this would be all the files added under `internals`

It would also simplify your work with git and tooling.

File: book/en/src/by-example/app_priorities.md (hunk 9 above)
Comment (jvanderk) [reply book/en/src/by-example/app_priorities.md:26]: Corrected.

File: book/en/src/by-example/app_priorities.md (hunk 4 above)
Comment (jvanderk) [reply book/en/src/by-example/app_priorities.md:7]: Fixed.

File: book/en/src/internals/conceptual_view.md (hunk 10 above)
Comment (jvanderk) [reply book/en/src/internals/conceptual_view.md:80]: Makes sense. I will try to remove these changes from this branch (not sure if that is correct terminology), open a new branch against the root of this branch and open a new pull request on that. Big adventure :-)

Comment (jvanderk) [reply book/en/src/internals/conceptual_view.md]: This pull request should now only contain editorial changes. The conceptual model related text and graphics will be put into another pull request. 

I am not sure if the history in *this* pull request will survive, if so if that is desirable or if that can be squashed out.

**Comment by burrbull** [comment]
`mdbook` supports SVG. You don't need to convert it to PNG. Just cut margins.
```md
![Conceptual execution model](./conceptual_execution_model.svg)
```

**Comment by jvanderk** [comment]
> `mdbook` supports SVG. You don't need to convert it to PNG. Just cut margins.
> 
> ```md
> 
> Conceptual execution model
> 
> ```

Made change and moved files to other pull request.

File: book/en/src/SUMMARY.md [hunk 11]
```diff
@@ -29,6 +29,7 @@
   - [v0.4.x to v0.5.x](./migration/migration_v4.md)
   - [RTFM to RTIC](./migration/migration_rtic.md)
 - [Under the hood](./internals.md)
+  - [Conceptual execution model](./internals/conceptual_view.md)
```
Comment (AfoHT) [inline book/en/src/SUMMARY.md:32]: I think this should go with the other PR :)

File: book/en/src/SUMMARY.md [hunk 12]
```diff
@@ -37,3 +38,6 @@
   <!--- [Ceiling analysis](./internals/ceilings.md)-->
   <!--- [Software tasks](./internals/tasks.md)-->
   <!--- [Timer queue](./internals/timer-queue.md)-->
+
+(XXX would it be a good idea to add a section on code organization,
+i.e. how to break a larger application into a set of files? XXX)
```
Comment (AfoHT) [inline book/en/src/SUMMARY.md:43]: Same here as above

File: book/en/src/by-example/hardware_tasks.md [hunk 13]
```diff
@@ -1,24 +1,26 @@
 # Hardware tasks
 
-At its core RTIC is using the hardware interrupt controller ([ARM NVIC on cortex-m][NVIC])
-to perform scheduling and executing tasks, and all tasks except `#[init]` and `#[idle]`
+At its core RTIC is using a hardware interrupt controller ([ARM NVIC on cortex-m][NVIC])
+to schedule and start execution of tasks. All tasks except `pre-init`, `#[init]` and `#[idle]`
```
Comment (AfoHT) [inline book/en/src/by-example/hardware_tasks.md:4]: Did we want some kind of link for the somewhat special `pre-init`, not RTIC specific, but cortex-m-rt `#[pre_init]`?

https://docs.rs/cortex-m-rt/latest/cortex_m_rt/attr.pre_init.html

**Comment by AfoHT** [comment]
Forgot that GitHub doesn't like that kind of move, Gerrit influences from work I guess xD

Will look into this during the day...
